ethdb/remotedb/remotedb.go

```diff
@@ -33,8 +33,9 @@ type Database struct {
 }
 
 func (db *Database) Has(key []byte) (bool, error) {
-	if _, err := db.Get(key); err != nil {
-		return false, nil
+	_, err := db.Get(key)
+	if err != nil {
+		return false, err
```

Comment:
This is a bit of a tricky on, It kinda depends on how the remote database works. If the database returns an error if we try to `Get` something thats not available, then the new logic does not work.